Sync v2-8-stable with v2-8-test to release 2.8.0

ephraimbuddy opened the pull request:
Time for `2.8.0rc1`!

dstandish committed:
Run triggers inline with dag test (#34642)

No need to have trigger running -- will just run them async.

(cherry picked from commit 7b37a785d0b74d1e83c7ce84729febffd6e26821)

theaadya committed:
improved visibility of tasks in ActionModal for taskInstance (#35810)

* change max box height in ActionModal.tsx

* changed max box height in ActionModal.tsx

(cherry picked from commit 9a1dceb031aa0ab44a7c996c267128bd4c61a5bf)

dstandish committed:
Use ExitStack to manage mutation of secrets_backend_list in dag.test (#34620)

Although it requires another indent, it's cleaner, and more importantly it makes sure that the mutation is undone after failure.

(cherry picked from commit 99b4eb769d2a3b6692de9c0d83ba64041abf5789)

vincbeck committed:
Implement `is_authorized_variable` in AWS auth manager (#35804)

(cherry picked from commit 3b3ebafdce440952d2406955de290092ca0e361d)

vishnucoder1 committed:
Relax mandatory requirement for start_date when schedule=None (#35356)

* Relax mandatory requirement for start_date when schedule=None

* Updated run_type in unit tests

* Added check for empty start_date and non empty schedule

* Fix the build failures

* Fix the build failures

* Update based on review comments

(cherry picked from commit 930f165db11e611887275dce17f10eab102f0910)

hussein-awala committed:
Add a public interface for custom weight_rule implementation (#35210)

* Add a public interface for custom weight_rule implementation

* Remove _weight_strategy attribute

* Move priority weight calculation to TI to support advanced strategies

* Fix loading the var from mapped operators and simplify loading it from task

* Update default value and deprecated the other one

* Update task endpoint API spec

* fix tests

* Update docs and add dag example

* Fix serialization test

* revert change in spark provider

* Update unit tests

(cherry picked from commit 3385113e277f86b5f163a3509ba61590cfe7d8cc)

hussein-awala committed:
Consolidate the call of change_state to fail or success in the core executors (#35901)

(cherry picked from commit ce7f043e15534c3d9ba6d59c3bb6b851e36a60b9)

potiuk committed:
Remove workaround for pymssql failing compilation with new Cython (#35924)

Recent Cython release caused pymssql package failures when they
were installed on ARM platform. This had been workarounded in
the #32748, but since pymssql as of 2.1.8 supports new Cython, we
can remove the workaround and bump the minimum version of pymsssql.

This also makes it possible to remove the whole MSSQL client section
from the image if we decide to - because this section will only install
the odbc client that has been pre-installed to support MSSQL as
metadata DB for Airflow core.

(cherry picked from commit 4f060a482c3233504e7905b3ab2d00fe56ea43cd)

yermalov-here committed:
Change dag grid overscroll behaviour to auto (#35717)

(cherry picked from commit d37b91c102856e62322450606474aebd74ddf376)

ephraimbuddy committed:
Revert "Prevent assignment of non JSON serializable values to DagRun.conf dict (#35096)" (#35959)

This reverts commit 84c40a7877e5ea9dbee03b707065cb590f872111.

(cherry picked from commit 4a7c7460bf1734b76497280f5a2adc3e30a7820c)

Taragolis committed:
Rename `Connection.to_json_dict` to `Connection.to_dict` (#35894)

(cherry picked from commit 7594b7a8eecd216dbaf31fcbd958ba22a97d6709)

ephraimbuddy committed:
Move `duckdb` & `pandas` import in tutorial DAG into task (#35964)

This improves the code as per best practices and avoids import
error if duckdb is not installed

(cherry picked from commit f0ba2dced92c767367aaf0fa3147942b4a576f92)

hussein-awala committed:
Use S3 hook instead of AwsGenericHook in AWS S3 FS (#35973)

(cherry picked from commit df36e7381ac50202568b339803fd8b78c6ed205e)

potiuk committed:
Mark daskexecutor provider as removed (#35965)

Following the discussion in airflow devlist we mark daskexecutor
as removed. See:
https://lists.apache.org/thread/fxv44cqqljrrhll3fdpdgc9h9fz5ghcy

(cherry picked from commit 9c1c9f450e289b40f94639db3f0686f592c8841e)

tirkarthi committed:
Add processor_subdir to import_error table to handle multiple dag processors (#35956)

* Add processor_subdir to import_error table to handle multiple dag processors.

* Fix migration table name.

(cherry picked from commit 1a3eeab76cdb6d0584452e3065aee103ad9ab641)

renzepost committed:
Fix airflow db shell needing an extra keypress to exit (#35982)

(cherry picked from commit cbb9c4f8ccadd5fbd01a2f0072343764eab03497)

potiuk committed:
Bump FAB to 4.3.10 (#35991)

(cherry picked from commit 9bcee9d439ada9104e702e090b20d5d1eeafa035)

potiuk committed:
Add feature to build "chicken-egg" packages from sources (#35890)

When we build on ci a pre-release version of provider, and we want
to include packages that have >= CURRENTLY_RELEASED_VERSION
we have to make sure that those packages are built from sources
during building of PROD image. Otherwise they will not be installable
on CURRENT_VERSION.dev0, CURRENT_VERSION.rc* etc.

Until we "Actually" release a provider we should have a way to
build such provider from sources.

This is the CI version of it, once we have it working we can also
apply it to the workflow that releases images to dockerhub.

(cherry picked from commit cf052dc64f00e851427a41a34ffe576fd39be51b)

uranusjr committed:
Pass conn ID to ObjectStoragePath via URI (#35913)

This enables an alternative ObjectStoragePath init syntax, using the
auth section in the URI to supply conn ID instead of a separate keyword
argument. The explicit keyword argument is honored if supplied.

(cherry picked from commit ab87cd02e4080df987a123d56cd9fa8c393a980d)

potiuk committed:
Switch "latest" image to point to newest supported Python version (#36003)

Following the lazy consensus to change the "latest" image
to point to "newest" Python version, we are changing the release
method to follow it.

https://lists.apache.org/thread/0oxnvct24xlqsj76z42w2ttw2d043oy3
(cherry picked from commit 4117f1b013323d851613ba7b69b3d987cf213ead)

potiuk committed:
Add support for chicken-egg providers to dockerhub release process (#36002)

(cherry picked from commit 8829d1732c7f210c9d2ab0cc20ebfb1861ae9f84)

mhenc committed:
[AIP-44] Introduce Pydantic model for LogTemplate (#36004)

(cherry picked from commit c26aa12bcc91429f0c2dd53e066a8480c00f822c)

vchiapaikeo committed:
Add multiselect to run state in grid view (#35403)

* Add multiselect to run state in grid view

* Fix tests

* Multiselect for run types, UI nits, refactor

* Fix tests, refactor

* Simplify multiselect value

* Nits and refactor

* Use arrays instead of serializing to csv

* Fix tests and global axios paramsSerializer to null

(cherry picked from commit 9e28475402a3fc6cbd0fedbcb3253ebff1b244e3)

jscheffl committed:
Change Trigger UI to use HTTP POST in web ui (#36026)

* Change Trigger UI to use HTTP POST in web ui, GET always shows trigger form
* Adjust tests to changed behavior of trigger handling, expects data submitted in POST

(cherry picked from commit f5d802791fa5f6b13b635f06a1ea2eccc22a9ba7)

tirkarthi committed:
Use dropdown instead of buttons when there are more than 10 retries in log tab (#36025)

* Use dropdown instead of buttons when there are more than 10 retries in log tab.

* Update airflow/www/static/js/dag/details/taskInstance/Logs/index.tsx

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

---------

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>
(cherry picked from commit fd0988369b3a94be01a994e46b7993e2d97b2028)

avkirilishin committed:
34058: Fix UI Grid error when DAG has been removed. (#36028)

(cherry picked from commit 549fac30eeefaa449df9bfdf58eb40a008e9fe75)

potiuk committed:
Limit Pytest-asyncio to < 0.23.1 (#36037)

Seems that pytest-asyncio 0.23.1 break our asyncio tests. We are
temporarily limiting it until
https://github.com/pytest-dev/pytest-asyncio/issues/703 is
solved or answered.

(cherry picked from commit 9845b40a755170eeee3537ac2c2676511ec54689)

potiuk committed:
Limit pytest-asyncio even more - to <0.23.0 (#36040)

Seems that the pytest-asyncio problem was already introduced by
0.23.0. In order to allow tests passing now, we should limit it
to below that version (follow up after #36037)

(cherry picked from commit cc2521cf6c01363f0e1c96bbd6ed0231406ecd63)

potiuk committed:
Add the section describing the security model of DAG Author capabilities (#36022)

* Add the section describing the security model of DAG Author capabilities

This change codifies and explains assumptions and decisions made by
Airflow maintainers with regards to capabilities of DAG Authors.

While DAG authors are pretty powerful and capable actors in Airflow,
they cannot do everything and Deployment Managers haw ways to restrict
their potential capabilities, especially in the context of influencing
other tasks and common components such as Scheduler, Webserver and
Triggerer.

This PR adds a chapter explaining those assumptions and decisions and
tell the Deployment Managers what responsibilities they have with
that regardsm and what mechanismes they currently have available to
limit capabilities of DAG Authors.

* Update docs/apache-airflow/security/security_model.rst

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>

* Update docs/apache-airflow/security/security_model.rst

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>

---------

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>
(cherry picked from commit 395ac463494dba1478a05a32900218988495889c)

potiuk committed:
Remove pytest-asyncio upper-binding limitatin (#36046)

Thahnks to our report and bisecting the reason (it was caused by
bad handling of the "test*.txt" handling) we can now remove the
upper-binding for pytest-asyncio as 0.23.2 version has been
released

(cherry picked from commit 55eac5e2c701be46e5ea5498b597297130edcd0e)

hduong-mwam committed:
Add XCom tab to Grid (#35719)

* Add XCom tab to Grid

* Combine showLogs and showXcom logic evaluation to isIndividualTaskInstance

* Remove link to /xcom page from UI grid view

* Use consistent naming to distinguish XcomCollection and XcomEntry

* Refactor boolean vars

(cherry picked from commit 77c01031d6c569d26f6fabd331597b7e87274baa)

andar9 committed:
Update supported-versions.rst (#36058)

I think the date 14.19.2023 is wrong--should it be 14.09.2023 (= September 14, 2023)?

(cherry picked from commit 55d81378b0fa7488bfdca68a4693821109c8fd1e)

romsharon98 committed:
Avoid crushing container when directory is not found on rm (#36050)

(cherry picked from commit 61fd166a4662d67bc914949f9cf07ceab7d55686)

pdebelak committed:
Update reset_user_sessions to work from either CLI or web (#36056)

This updates the `reset_user_sessions` method to not class `flash` if
it is not being run within the web context but to log the warning
message instead. Without this change, `flask fab reset-password` would
fail if a warning message needed to be printed.

(cherry picked from commit 7ececfdb2183516d9a30195ffcd76632167119c5)

ephraimbuddy committed:
Update RELEASE_NOTES.rst

tirkarthi committed:
Fix gantt chart queued duration when queued_dttm is greater than start_date for deferred tasks. (#35984)

* Fix gantt chart queued duration when queued_dttm is greater than start_date for deferred tasks.

* Merge hasQueuedDttm and validQueuedDttm.

(cherry picked from commit 0376e9324af7dfdafd246e31827780e855078d68)

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -41,6 +41,8 @@ Significant Changes
 """"""""""""
 - AIP-58: Add Airflow ObjectStore (AFS) (`AIP-58 <https://github.com/apache/airflow/pulls?q=is%3Apr+is%3Amerged+label%3AAIP-58+milestone%3A%22Airflow+2.8.0%22>`_)
+- Add XCom tab to Grid (#35719)
+- Add a public interface for custom ``weight_rule`` implementation (#35210)
```
We decided to remove it didn't we?

ephraimbuddy replied:
Yeah see: https://github.com/apache/airflow/pull/36065#issuecomment-1840504204

ephraimbuddy replied:
Reverted

marekhanus committed:
Add read access to pools for viewer role (#35352)

Cluster Activity UI shows `FAILED TO FETCH DATA` error in pools section. Users with Viewer role do not have permission to read pools.

(cherry picked from commit 8ae67a04e9a5982b30e46f9ffd3ec3b895bccd79)

potiuk committed:
Replace architecture diagram of Airflow with diagrams-generated one (#36035)

The architecture diagram of Airflow has been long time outdated.

This is an attempt to generate it using generated diagrams using
Python's diagrams library (already used by some tools in our
ecosystem).

(cherry picked from commit 5dfee8b992e23370937c28052ebca231c361da71)

uranusjr committed:
Account for change in UTC offset when calculating next schedule (#35887)

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>
(cherry picked from commit c4549d721f3b8127e023f1953378484f02e5919b)

hussein-awala committed:
Revert "Add a public interface for custom weight_rule implementation (#35210)" (#36066)

This reverts commit 3385113e277f86b5f163a3509ba61590cfe7d8cc.

(cherry picked from commit f60d458dc08a5d5fbe5903fffca8f7b03009f49a)

josh-fell committed:
Add support for tabs (and other UX components) to docs (#36041)

(cherry picked from commit 58e264c83fed1ca42486302600288230b944ab06)

potiuk committed:
Fix optimization of PROD image building side-effect

The change #35856 optimized waiting time before PROD image builds
start - rather than waiting for full constratints generation, the
PROD image building just used source constraints generated right
after building the CI image quickly. This is fine for main because there
we install airflow and packages using constraints from sources, but
for release branches we use the provider constraints - in order
to be able to install providers from PyPI rather than from sources.

This means that we have to wait for constraints generation to
complete before we start building PROD images - because we need to
download the constraints generated there to use them.

Unfortunately GitHub Actions do not have conditional dependencies
depending on where the workflow is run  - so instead we have to
effectively duplicate PROD build steps and skip steps in them instead.

bolkedebruin committed:
Allow storage options to be passed (#35820)

This allows storage options to be passed to the fsspec backend as part
of the kwargs to ObjectStorage.

---------

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit aba58adb83435e96e66027de5163756301a6ddf5)

hussein-awala reviewed (approved):
Looks good to me, LGTM

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -63,6 +64,14 @@ New Features
 
 Improvements
 """"""""""""
+- Account for change in UTC offset when calculating next schedule (#35887)
```
This is a bug fix isn't it?

ephraimbuddy replied:
It's part of AIP-39

ephraimbuddy replied:
But I will move it

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -146,6 +165,10 @@ Bug Fixes
 
 Misc/Internal
```
For later discussion
I don't think `Misc/Internal` is good title.
If it's Internal then we are marking with `changelog:skip` and it will not appear in the change log.

potiuk replied:
Agree. `Misc` should be for those changes that do not introduce new features or bug fixes - but can have impact on what the user experiences. Good example is bumping version of FAB or Pydantic - both of which are neither features nor bug fixes, but they might cause some issues for the user and the user needs to be able to learn from the release notes that "Something" has changed. Similalry daskexcutor removal or `pyarrow_hotfix`.

So I think the only problem here is the name (which IMHO should be just `Misc`).

ephraimbuddy replied:
Let me change it today :)

potiuk commented on RELEASE_NOTES.rst:
```diff
@@ -122,6 +131,16 @@ Improvements
+- Avoid crushing container when directory is not found on rm (#36050)
+- Update ``reset_user_sessions`` to work from either CLI or web (#36056)
+- Fix UI Grid error when DAG has been removed. (#36028)
+- Use dropdown instead of buttons when there are more than 10 retries in log tab (#36025)
```
This one could also be classified as feature.
